go/service/chat_local_test.go

```diff
@@ -325,6 +328,32 @@ func TestChatGetThreadLocalMarkAsRead(t *testing.T) {
 		t.Fatalf("no TEXT message in returned inbox")
 	}
 
+	// Do a get thread local without requesting marking as read first. This
+	// should cause HybridConversationSource to cache the thread. Then we do
+	// another call requesting marking as read before checking if the thread is
+	// marked as read. This is to ensure that when the query requests for a
+	// mark-as-read, and the thread gets a cache hit, the
+	// HybridConversationSource should not just return the thread, but also send
+	// a MarkAsRead RPC to remote. (Currently this is done in
+	// HybridConversationSource.Pull)
+	//
+	// TODO: This doesn't make sense! In integration tests, this isn't necessary
+	// since a Pull() is called during PostLocal (when populating the Prev
+	// pointers).  However it seems in this test, it doesn't do so. This first
+	// GetThreadLocal always gets a cache miss, resulting a remote call. If
+	// PostLocal had worked like integration, this shouldn't be necessary. We
+	// should find out where the problem is and fix it! Although after that fix,
+	// this should probably still stay here just in case.
+	_, err = ctc.as(t, users[0]).chatLocalHandler().GetThreadLocal(context.Background(), chat1.GetThreadLocalArg{
```

Comment:
So the answer to this question isn't the best, but right now the caching of the messages that  come from `PostLocal` only happens because we get a Gregor OOBM for it. That is, we send the message, the server then sends an OOBM telling us we sent the message, and we cache that. We probably want to just cache the message directly from `PostLocal`.

The integration test works because Gregor messages work there, and not in this unit test.